app/controllers/v0/sessions_controller.rb

```diff
@@ -34,15 +33,16 @@ def new
 
       StatsD.increment(STATSD_SSO_NEW_KEY, tags: ["context:#{type}", VERSION_TAG])
       Rails.logger.info("SSO_NEW_KEY, tags: #{["context:#{type}", VERSION_TAG]}")
-      url = url_service.send("#{type}_url")
+      helper = url_service
```

Comment:
the url_service method could instead be something like this so it can be used multiple times in an action without needing to be assigned to a variable
```
def url_service(previous_saml_uuid = nil)
      @url_service ||= SAML::URLService.new(saml_settings,
                           session: @session_object,
                           user: current_user,
                           params: params,
                           previous_saml_uuid: previous_saml_uuid)
    end
```

Reply:
thats a brilliant idea!  I just created this issue to make that change in the future https://github.com/department-of-veterans-affairs/va.gov-team/issues/13231